### PR TITLE
Feature/network

### PR DESCRIPTION
**Pull request by florianow**

### CODE REVIEW

File: kit/azure/networking/locals.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    source_port_ranges                         = null
+    destination_port_range                     = null
+    destination_port_ranges                    = null
+    source_address_prefix                      = null
+    source_address_prefixes                    = null
+    source_application_security_group_ids      = null
+    destination_address_prefix                 = null
+    destination_address_prefixes               = null
+    destination_application_security_group_ids = null
+  }
```
Comment (felixzieger) [inline kit/azure/networking/locals.tf:32]: Q: What effect does this rule have? Source and destination ranges are set to null. So no effect?

**Review by felixzieger — APPROVED** [review]
As discussed, merging the subscriptions dir into tenants makes sense.

Adding a readme to the building blocks dir with something like
```
Building Blocks are not standard kit modules.

After applying a building block with `collie kit apply"`, we recommend to rename the created directory. This allows you to apply the same building block kit module again later.
```

File: kit/azure/networking/locals.tf (hunk 1 above)
Comment (florianow) [reply kit/azure/networking/locals.tf]: https://github.com/meshcloud/collie-hub/blob/3bbb0194ab0d31caf0f2f1958863a9f8b33f515b/kit/azure/networking/resources.network.tf#L250 you could use every possible value in the resource.

File: kit/azure/networking/resources.network.tf [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    }
+  }
+}
+
+resource "azurerm_network_security_rule" "mgmt" {
+  count = length(local.merged_mgmt_nsg_rules)
+
+  resource_group_name                        = azurerm_resource_group.hub_resource_group.name
+  network_security_group_name                = azurerm_network_security_group.mgmt.name
+  priority                                   = 100 + 100 * count.index
```
Comment (felixzieger) [inline kit/azure/networking/resources.network.tf:255]: d: This means the index of the rule in local.merged_mgmt_nsg_rules determines the priority.   

They are defined via a [merge](https://developer.hashicorp.com/terraform/language/functions/merge)
```hcl
merged_mgmt_nsg_rules = flatten([
    for nsg in var.management_nsg_rules : merge(nsg)
  ])
```

This should be made clear.

I couldn't find out if the order is determined deterministically. It's not documented on the merge function documentation page. And it's not stated in the README.

File: kit/azure/networking/locals.tf (hunk 1 above)
Comment (felixzieger) [reply kit/azure/networking/locals.tf:32]: That makes sense. I have one more question about the priority of rules, see separate comment.

File: kit/azure/networking/resources.network.tf (hunk 2 above)
Comment (felixzieger) [reply kit/azure/networking/resources.network.tf:255]: But it still means that priority is determined by the index in some list.

So for [a, b] means 
- a has prio 100 and 
- b has prio 200.

However if you do [b, a] you get the reverse: 
- a has prio 200 and 
- b has prio 100.

If users know that the position of the rule determines the prio, I'm fine with it. But this is not documented

Comment (florianow) [reply kit/azure/networking/resources.network.tf:255]: Good point, u are right. At the end its a filled pre-configuration should point out the possibilities and be adapted for further processing. I will adding a comment, thanks.

**Pull request merged**